### DISCUSSION TITLE
How I can eval model after each epoch end?

### DISCUSSION THREAD

**Discussion by SherlockHolmes221** [fredzzhang/upt#48] [Q&A] [answered]
How I can eval model after each epoch end? Thanks

**Comment by fredzzhang** [comment, marked as answer]
Hi @SherlockHolmes221,

The evaluation code is implemented for a single GPU, without distributed data parallel. So, it wasn't included during training. However, all checkpoints after each epoch will be cached by default in the directory you specified. Each checkpoint is named as `ckpt_steps_epochs`. You can run evaluation on the checkpoints you are interested in.

Fred.

**Comment by SherlockHolmes221** [comment]
Thx

**Comment by leijue222** [comment]
It is wise not to include evaluation code in the training process. VCOCO's evaluation code is so slow that it takes me 20 minutes for each evaluation. I was wondering if VCOCO's evaluation method could be improved to speed things up.
